src/mame/drivers/tim011.cpp

```diff
@@ -79,27 +77,34 @@ INPUT_PORTS_END
 
 void tim011_state::machine_reset()
 {
+	m_scroll = 0;
 	// motor is actually connected on TXS pin of CPU
-	m_floppy0->mon_w(0);
-	m_floppy1->mon_w(0);
-	m_floppy2->mon_w(0);
-	m_floppy3->mon_w(0);
-}
-
-void tim011_state::video_start()
-{
+	for (auto &drive : m_floppy)
+	{
+		if (drive->get_device())
+			drive->get_device()->mon_w(0);
+	}
 }
 
 uint32_t tim011_state::screen_update_tim011(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
+	u16 addr = 0;
+	for (u8 x = 0; x < 512/4; x++)
+	{
+		for (int y = cliprect.top(); y <= cliprect.bottom(); y++)
+		{
+			int horpos = x*4;
+			u8 code = m_vram[addr++];
+			u8 real_y = (y - m_scroll) & 0xff;
+			bitmap.pix(real_y, horpos++) =  (code >> 0) & 0x03;
+			bitmap.pix(real_y, horpos++) =  (code >> 2) & 0x03;
+			bitmap.pix(real_y, horpos++) =  (code >> 4) & 0x03;
+			bitmap.pix(real_y, horpos++) =  (code >> 6) & 0x03;
+		}
```

Comment:
This won’t actually work – you’re treating the clipping rectangle bounds as if they apply to video RAM organisation when they actually apply to screen positions.

You need to reorganise this so the loop counter is `real_y` and then calculate the address offset corresponding to that row considering the scroll register value.